projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1783,6 +1799,36 @@ public function has_paid_plan( $refresh_from_wpcom = false ) {
 		}
 		return $has_paid_plan;
 	}
+
+	/**
+	 * Check if we have a Jetpack Social Advanced plan.
+	 *
+	 * @param bool $refresh_from_wpcom Whether to force refresh the plan check.
+	 *
+	 * @return bool True if we have an Advanced plan, false otherwise.
+	 */
+	public function has_advanced_plan( $refresh_from_wpcom = false ) {
+		static $has_advanced_plan = null;
+		if ( ! $has_advanced_plan ) {
```

Comment:
I think we'd need to check against `null` as otherwise `false` would keep triggering the plan check.

```suggestion
		if (  null === $has_advanced_plan ) {
```

Should we also check for `refresh_from_wpcom` being true? Otherwise it will always return the cached value.